components/user/UserMilestone.js

```diff
@@ -12,11 +12,13 @@ export default function UserMilestone({ milestone, isGoal, manage }) {
   const [date, setDate] = useState(milestone.date);
 
   useEffect(() => {
-    const formattedDate = shortenDate({
-      date: milestone.date,
-      formatStyle: milestone.dateFormat,
-    });
-    setDate(formattedDate);
+    if(milestone.date) {
+      const formattedDate = shortenDate({
+        date: milestone.date,
+        formatStyle: milestone.dateFormat,
+      });
+      setDate(formattedDate);
+    }
   }, [milestone, milestone.dateFormat]);
```

Comment:
```suggestion
  }, [milestone.date, milestone.dateFormat]);
```
There is no need to use ```milestone.dateFormat``` if we are adding ```milestone``` we can either use ```milestone``` as dependency or we can add ```milestone.date``` and ```milestone.dateFormat```